src/foam/u2/view/UnstyledTableView.js

```diff
@@ -75,7 +75,7 @@ foam.CLASS({
       expression: function(columns, of, editColumnsEnabled, selectedColumnNames, allColumns) {
         if ( ! of ) return [];
         var cols;
-        if ( ! editColumnsEnabled ) cols = columns;
+        if ( ! editColumnsEnabled ) cols = of.getAxiomByName('tableColumns').columns || allColumns;
         else cols = selectedColumnNames;
```

Comment:
for some reason 'columns' expression is unreachable that's why code moved here

Reply:
so I'm wondering if we could change unitPropName to array of names or maybe there is a better solution

Reply:
The 'columns' expression would not be called if that field's value were explicitly set. Why an array?